core/server/middleware/error-handler.js

```diff
@@ -35,64 +37,30 @@ _private.parseStack = function (stack) {
     );
 };
 
-_private.handleHTMLResponse = function handleHTMLResponse(err, req, res) {
-    return function handleHTMLResponse() {
-        var availableTheme = config.get('paths').availableThemes[req.app.get('activeTheme')] || {},
-            defaultTemplate = availableTheme['error.hbs'] ||
-                path.resolve(config.get('paths').adminViews, 'user-error.hbs') ||
-                'error';
-
-        res.render(defaultTemplate, {
-            message: err.message,
-            code: err.statusCode,
-            stack: _private.parseStack(err.stack)
-        }, function renderResponse(err, html) {
-            if (!err) {
-                return res.send(html);
-            }
-
-            // And then try to explain things to the user...
-            // Cheat and output the error using handlebars escapeExpression
-            return res.status(500).send(
-                '<h1>' + i18n.t('errors.errors.oopsErrorTemplateHasError') + '</h1>' +
-                '<p>' + i18n.t('errors.errors.encounteredError') + '</p>' +
-                '<pre>' + hbs.handlebars.Utils.escapeExpression(err.message || err) + '</pre>' +
-                '<br ><p>' + i18n.t('errors.errors.whilstTryingToRender') + '</p>' +
-                err.statusCode + ' ' + '<pre>' + hbs.handlebars.Utils.escapeExpression(err.message || err) + '</pre>'
-            );
-        });
-    };
-};
-
-/**
- * @TODO: jsonapi errors format
- */
-_private.handleJSONResponse = function handleJSONResponse(err, req, res) {
-    return function handleJSONResponse() {
-        res.json({
-            errors: [{
-                message: err.message,
-                errorType: err.errorType,
-                errorDetails: err.errorDetails
-            }]
-        });
-    };
-};
-
 /**
- * @TODO: test uncaught exception (wrap into custom error!)
+ * Get an error ready to be shown the the user
+ *
  * @TODO: support multiple errors
  * @TODO: decouple req.err
  */
-module.exports = function errorHandler(err, req, res, next) {
+_private.prepareError = function prepareError(err, req, res) {
+    /*jshint unused:false */
     if (_.isArray(err)) {
         err = err[0];
     }
 
     if (!(err instanceof errors.GhostError)) {
-        err = new errors.GhostError({
-            err: err
-        });
+        // We need a special case for 404 errors
+        if (err.statusCode && err.statusCode === 404) {
```

Comment:
originally https://github.com/TryGhost/Ghost/pull/7542#discussion_r82830112
is see that you have changed it already.

How about putting the 404 check into the GhostError class?